kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/utils/KafkaUtils.java

```diff
@@ -97,16 +97,34 @@ public static String ConsumeMessageWithTestClients(String deployNamespace, Strin
 
         kubeClient().getClient().load(file).inNamespace(deployNamespace).create();
         String podName = getPodNameByLabel(deployNamespace, "app", Constants.KAFKA_CONSUMER_CLIENT_LABEL, timeoutMilliseconds);
-        await().ignoreException(KubernetesClientException.class).atMost(Duration.ofMillis(timeoutMilliseconds)).until(() -> {
-            if (kubeClient().getClient().pods().inNamespace(deployNamespace).withName(podName).get() != null) {
-                var log = kubeClient().logsInSpecificNamespace(deployNamespace, podName);
-                return log.contains(" - " + (numOfMessages - 1));
-            }
-            return false;
-        });
+        await().alias("Consumer waiting to receive messages").ignoreException(KubernetesClientException.class).atMost(Duration.ofMillis(timeoutMilliseconds))
+                .until(() -> {
+                    if (kubeClient().getClient().pods().inNamespace(deployNamespace).withName(podName).get() != null) {
+                        var log = kubeClient().logsInSpecificNamespace(deployNamespace, podName);
+                        return log.contains(" - " + (numOfMessages - 1));
+                    }
+                    return false;
+                });
         return kubeClient().logsInSpecificNamespace(deployNamespace, podName);
     }
 
+    /**
+     * Admin test client.
+     *
+     * @param deployNamespace the deploy namespace
+     * @param bootstrap the bootstrap
+     * @return the pod name
+     */
+    public static String AdminTestClient(String deployNamespace, String bootstrap) {
+        InputStream file = replaceStringInResourceFile("kafka-admin-template.yaml", Map.of(
+                "%BOOTSTRAP_SERVERS%", bootstrap));
+
+        kubeClient().getClient().load(file).inNamespace(deployNamespace).create();
+        String podName = getPodNameByLabel(deployNamespace, "app", Constants.KAFKA_ADMIN_CLIENT_LABEL, Duration.ofSeconds(10).toMillis());
+        DeploymentUtils.waitForDeploymentReady(deployNamespace, Constants.KAFKA_ADMIN_CLIENT_LABEL);
```

Comment:
I think you are running a deployment (which does nothing except sleep), the exec'ing into the pod to invoke CLI commands.  Is that right?  What's the advantage to this approach over just using individual Kubernetes jobs? (which seems less surprising).

Reply:
It is counterintuitive/violates least surprise:

- a long running pod which is actually doing nothing except sleeping .. pointless
- `kubectl exec` seems to exist to support [debugging use cases](https://kubernetes.io/docs/tasks/debug/debug-application/get-shell-running-container/) so it seems weird to use it to provide functionality.

I think kubernetes jobs would be a better approach.

Reply:
Thanks for the info! I didn't know that. I will work on that now.